hack/generated/controllers/generic_controller.go

```diff
@@ -535,3 +601,69 @@ func (gr *GenericReconciler) Patch(
 		Name:      data.metaObj.GetName(),
 	}, data.metaObj)
 }
+
+// isOwnerReady returns true if the owner is ready or if there is no owner required
+func (gr *GenericReconciler) isOwnerReady(ctx context.Context, data *ReconcileMetadata) (bool, error) {
+	_, err := gr.ResourceResolver.GetOwner(ctx, data.metaObj)
+	if err != nil {
+		switch causeErr := errors.Unwrap(err).(type) {
```

Comment:
I think this should be using either `errors.Cause` or `.As`. `errors.Unwrap` only peels off one layer, so you're assuming too much about the call stack of `GetOwner` - if it gets changed to add another layer of context at some point this will stop working.

Reply:
I used `Cause` originally but the problem is that it unwraps to the first error that doesn't implement the `Cause` interface - but since I made my error implement that interface it bypassed that.

`As` sounds right though I think.